src/main/java/com/shipmentEvents/handlers/badFile.java

```diff
@@ -0,0 +1,68 @@
+package com.shipmentEvents.handlers;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+
+public class badFile {
+    
+    public HashMap processShipment(String shipmentId, HashMap shipmentData) {
+        // Example of bad/no input validation
+        if (shipmentId == "" || shipmentData == null) {
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**: An equality operator (== or !=) is used to compare strings, which matches strings based on address.

**Fix**: Do content comparison using the equals() method to compare the values of the strings.